### PR TITLE
3 ➡️ 4

### PR DESCRIPTION
**Pull request by mjcarroll**
Forward-port ign-gazebo3 to ign-gazebo4.

Comparison:

https://github.com/ignitionrobotics/ign-gazebo/compare/ign-gazebo4...ign-gazebo3

### CODE REVIEW

File: tutorials/migrating_ardupilot_plugin.md [hunk 1]
```diff
@@ -24,7 +22,7 @@ documentation](https://ardupilot.org/dev/docs/using-gazebo-simulator-with-sitl.h
 As context to understand what we're migrating, here's a system diagram for how
 the ArduPilot Gazebo plugin works is used:
 
-<img src="https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/master/tutorials/files/ardupilot_diagram.png"/>
+<img src="https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/ign-gazebo3/tutorials/files/ardupilot_diagram.png"/>
```
Comment (chapulina) [inline tutorials/migrating_ardupilot_plugin.md:25]:
```suggestion
<img src="https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/ign-gazebo4/tutorials/files/ardupilot_diagram.png"/>
```

File: tutorials/migrating_ardupilot_plugin.md [hunk 2]
```diff
@@ -226,10 +224,10 @@ To better understand the ECS pattern as it is used in Ignition, it's helpful to
 learn about the EntityComponentManager (ECM), which is responsible for managing
 the ECS graph. A great resource to understand the logic under the hood of the
 ECM is the `SdfEntityCreator` class
-([header](https://github.com/ignitionrobotics/ign-gazebo/blob/master/include/ignition/gazebo/SdfEntityCreator.hh),
-[source](https://github.com/ignitionrobotics/ign-gazebo/blob/master/src/SdfEntityCreator.cc)).
+([header](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo3/include/ignition/gazebo/SdfEntityCreator.hh),
```
Comment (chapulina) [inline tutorials/migrating_ardupilot_plugin.md:227]:
```suggestion
([header](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/include/ignition/gazebo/SdfEntityCreator.hh),
```

File: tutorials/migrating_ardupilot_plugin.md [hunk 3]
```diff
@@ -226,10 +224,10 @@ To better understand the ECS pattern as it is used in Ignition, it's helpful to
...
+[source](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo3/src/SdfEntityCreator.cc)).
```
Comment (chapulina) [inline tutorials/migrating_ardupilot_plugin.md:228]:
```suggestion
[source](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/src/SdfEntityCreator.cc)).
```

File: tutorials/video_recorder.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+
+## Video recorder configurations
+
+A few video recorder parameters can be specified using GUI configurations, see
+the [GUI Configuration](gui_config.html) tutorial for more information.
+If you launched Ignition Gazebo with the
+`video_record_dbl_pendulum.sdf` demo world, the GUI configurations are embedded
+in the world SDF file so you will need to download a copy of the
+[sdf file](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/ign-gazebo3/examples/worlds/video_record_dbl_pendulum.sdf).
```
Comment (chapulina) [inline tutorials/video_recorder.md:41]:
```suggestion
[sdf file](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/ign-gazebo4/examples/worlds/video_record_dbl_pendulum.sdf).
```

**Review by chapulina — APPROVED** [review]
Just noticed a few links in tutorials that need to be updated. Besides this, LGTM with happy CI.

**Comment by mjcarroll** [comment]
Bionic CI is `139 - PERFORMANCE_each (Failed)`

Homebrew is: 
```
INTEGRATION_level_manager.test_ran
INTEGRATION_pose_publisher_system.test_ran
```

So I think we're good here.

**Pull request merged**